test/option-immutableMode.html

```diff
@@ -0,0 +1,167 @@
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+
+<html>
+    <head>
+        <meta charset="utf-8">
+        <meta name="viewport" content="width=device-width, initial-scale=1" />
+        <script src="lib/simpleRequire.js"></script>
+        <script src="lib/config.js"></script>
+        <script src="lib/jquery.min.js"></script>
+        <script src="lib/facePrint.js"></script>
+        <script src="lib/testHelper.js"></script>
+        <!-- <script src="ut/lib/canteen.js"></script> -->
+        <link rel="stylesheet" href="lib/reset.css" />
+    </head>
+    <body>
+        <style>
+        </style>
+
+
+        <div id="main_immutableMode_obj_true"></div>
+        <div id="main_immutableMode_param"></div>
+ 
+        <script>
+            function makeBasicOption(opt) {
+                return {
+                    xAxis: {
+                        type: 'category'
+                    },
+                    yAxis: {},
+                    legend: {},
+                    tooltip: {},
+                    dataZoom: [{
+                        type: 'slider'
+                    }, {
+                        type: 'inside'
+                    }],
+                    series: [{
+                        id: 'a',
+                        name: 'aa',
+                        type: 'line',
+                        data: [['a11', 22], ['a33', 44]]
+                    }, {
+                        id: 'b',
+                        name: 'bb',
+                        type: 'line',
+                        data: [['a11', 55], ['a33', 77]]
+                    }, {
+                        id: 'c',
+                        name: 'cc',
+                        type: 'line',
+                        data: [['a11', 66], ['a33', 100]]
+                    }, {
+                        id: 'd',
+                        name: 'dd',
+                        type: 'line',
+                        data: [['a11', 99], ['a33', 130]]
+                    }, {
+                        name: 'no_id',
+                        type: 'line',
+                        data: [['a11', 130], ['a33', 160]]
+                    }]
+                };
+            }
+        </script>
+
+
+        <script>
+        require(['echarts'], function (echarts) {
+            const option = makeBasicOption();
+
+            const opts =  { notMerge: true, lazyUpdate: false, immutableMode: true };
+
+            var chart = testHelper.create(echarts, 'main_immutableMode_obj_true', {
+                title: [
+                    'immutableMode: true',
+                    'click "setOption": "TODO: add immutableMode test case"',
```

Comment:
This test is just a placeholder for now (I'm currently testing in our app), but I'll come back to this and add proper examples and documentation about how/when to use `immutableMode: true` with [setOption](https://echarts.apache.org/en/api.html#echartsInstance.setOption)

Reply:
Running all existing visual regression tests (`npm run test:visual`), so far all the 'line' test cases have passed!

![image](https://github.com/sjcobb/echarts/assets/9369625/8558f219-1b3f-4c10-aa66-8f14f51a735d)

Also Jest unit tests are all passing (`npm run test`)

Reply:
All relevant existing visual regression tests are passing! A few related to geo map data failed but those also fail for me locally on master (seems to be an issue with connecting to an external URL). Still working on creating a new test case to for immutableMode specifically